src/app/register/_components/EmailPasswordBox.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+
+import * as React from "react";
+import { Card, CardContent } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+
+interface EmailPasswordBoxProps {
+  setPasswordMatch: (isMatch: boolean) => void;
+  setPasswordValid: (isValid: boolean) => void;
+  setEmailValid: (isValid: boolean) => void;
+  setEmailAlreadyReg: (isValid: boolean) => void;
+}
+
+export default function EmailPasswordBox(props: EmailPasswordBoxProps) {
+  const checkConfirmPassword = (props: EmailPasswordBoxProps) => {
+    const password = document.getElementById("Password") as HTMLInputElement;
+    const confirmPassword = document.getElementById(
+      "Confirm Password",
+    ) as HTMLInputElement;
+    if (!!confirmPassword && !!password && confirmPassword.value != "") {
+      if (password.value == confirmPassword.value) {
+        (document.getElementById("message") as HTMLInputElement).style.color =
+          "green";
+        (document.getElementById("message") as HTMLInputElement).innerHTML = "";
+        props.setPasswordMatch(true);
+      } else {
+        (document.getElementById("message") as HTMLInputElement).style.color =
+          "red";
+        (document.getElementById("message") as HTMLInputElement).innerHTML =
+          "not matching";
+        props.setPasswordMatch(false);
+      }
+    }
+  };
```

Comment:
use `useRef` instead of getElementById

Reply:
ละ Notice message ใช้ getElementById เหมือนเดิมมั้ยอะ

Reply:
nothing should use getElementById

Reply:
**Always think that you might do something very wrong if you need to use anything start with `document` in react.**